### DISCUSSION TITLE
TIF to PDF problem

### DISCUSSION THREAD

**Discussion by filippesic** [ImageMagick/ImageMagick#2136] [Help]
Hello guys, I have this strange issue with converting about 330 .tif files, which is just over 4GB combined to single .pdf file. I successfully converted previous ~300 files, but the folder was 3.95GB, so that's why I think that's the problem. I tried so many options from Image Magick forums and nothing changes. The end file is basically always identical to the previous take. 330 .tif pictures and it always corrupts on 311 even on other .tif collections. It's totally constant.

Any help is extremely appreciated, I'm kinda desperate at this point. 

specs: notebook intel 6c/12t, 16GB of RAM, M.2 SSD and 1050Ti GPU

**Comment by urban-warrior** [comment]
Have you read https://imagemagick.org/script/architecture.php#tera-pixel?  Try this command:

`magick -monitor -limit memory 2GiB -limit map 4GiB -define registry:temporary-path=./ *.tif image.pdf`

Does that command work for you?

**Reply by filippesic** [reply]
~~Where is that temporary-path suppose to point? Just a different partition?~~ I'm on Windows I didn't mention that.

edit: I got it. I'm trying it now.

**Reply by filippesic** [reply]
`Load/Image/NEW_1847[00969.tif]: 3455 of 3456, 100% complete
magick.exe: Unknown field with tag 40092 (0x9c9c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
magick.exe: Unknown field with tag 40093 (0x9c9d) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
Load/Image/NEW_1847[00970.tif]: 3455 of 3456, 100% complete
magick.exe: Unknown field with tag 40092 (0x9c9c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
magick.exe: Unknown field with tag 40093 (0x9c9d) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
Load/Image/NEW_1847[00971.tif]: 3455 of 3456, 100% complete
magick.exe: Unknown field with tag 40092 (0x9c9c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
magick.exe: Unknown field with tag 40093 (0x9c9d) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
Load/Image/NEW_1847[00972.tif]: 3455 of 3456, 100% complete
magick.exe: Unknown field with tag 40092 (0x9c9c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
magick.exe: Unknown field with tag 40093 (0x9c9d) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
Load/Image/NEW_1847[00973.tif]: 3455 of 3456, 100% complete
magick.exe: Unknown field with tag 40092 (0x9c9c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
magick.exe: Unknown field with tag 40093 (0x9c9d) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
Load/Image/NEW_1847[00974.tif]: 3455 of 3456, 100% complete
magick.exe: Unknown field with tag 40092 (0x9c9c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
magick.exe: Unknown field with tag 40093 (0x9c9d) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/1037.
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete
resize image[image.pdf]: 1 of 2, 100% complete`

The problem is the same. 312 pages and the rest is just blank. I pasted just a part of the log here. This "1 of 2" also didn't change at all. I also tried on Windows VM with 24 threads and 47GB of RAM and it's identical.
I'm not sure if I'm making some stupid mistake, this is what I'm using:

`magick -density 231x231 SOURCE_DIR_NAME/*.tif OUTPUT_DIR_NAME/singlePDF.pdf`

```
C:\Users\Administrator\Downloads>magick -list resource
Resource limits:
  Width: 214.748MP
  Height: 214.748MP
  Area: 51.0315GP
  List length: unlimited
  Memory: 23.7634GiB
  Map: 47.5268GiB
  Disk: unlimited
  File: 1536
  Thread: 24
  Throttle: 0
  Time: unlimited
```

**Comment by urban-warrior** [comment]
Your command does not include the limits nor the temporary path.  We tried this command on a 1000 frame TIF image and the command completed without complaint:

```
magick -monitor -limit memory 2GiB -limit map 4GiB -define registry:temporary-path=./ -density 231x231 SOURCE_DIR_NAME/*.tif OUTPUT_DIR_NAME/singlePDF.pdf
```

This is on a Linux box.  The final image was viewable by evince suggesting it was a valid PDF image.

Perhaps its a Windows issue or an issue with your particular image.  Can you post a link to your image so we can download it and reproduce the problem?  What version of libTIFF are you using?  What does `identify -list format` say for the TIFF format?

**Reply by filippesic** [reply]
My apologies for late answer. I will try a your command on my files later this evening on the VM. I would provide a link to the files, but it's not mine, it's for a company that I work currently. It's basically some old black and white books, lots of them, usually about 4.2GB and 300-350 files per book. Doesn't seem even remotely demanding or complicated for image magick, but I'm new to this so I thought I did something wrong.
And also, thanks a lot for still trying to help me. It sincerely means a lot to me.

**Reply by filippesic** [reply]
```
     TIFF* TIFF      rw+   Tagged Image File Format (LIBTIFF, Version 4.1.0)
   TIFF64* TIFF      rw+   Tagged Image File Format (64-bit) (LIBTIFF, Version 4.1.0)

```
identify -list format says this and 4.1.0 is the latest version right? I read something about .tiff how it doesn't support files above 4GB, but that was long time ago for much older versions than 4.1.0 correct? I tried your command like this:
`$ magick -monitor -limit memory 2GiB -limit map 4GiB -define registry:temporary-path=/f/ -density 231x231 NEW_1847/*.tif singlePdfTest.pdf
`
And the problem persists. No matter if I tried it on my notebook that I originally wrote about, the strong VM machine or my home PC with Ryzen 1600 and 8GB of RAM and about 85GB of free SSD space(/f/ partition), the end single .pdf file is always identical in size, it doesn't change even a single digit. `4,454,272KB` or `4.24GB` which is equal to all 331 .tif files, but after 312th page it's just blank in that .pdf file.
Should I try with VirtualBox Ubuntu maybe?

**Comment by urban-warrior** [comment]
Run your command with the latest ImageMagick release, 7.0.10-19.  We added additional checks to report if there are any I/O exceptions.  In the mean-time we created a 5GB file with this command:

```
magick -monitor -limit memory 2GiB -limit map 4GiB -define registry:temporary-path=/opt/tmp -density 231x231 SOURCE_DIR_NAME/*.tif OUTPUT_DIR_NAME/image.pdf
```

The command completes for us and generates expected results.  Make sure your temporary path exists, has plenty of free space, and has proper permissions.  If the command fails for you under Windows, try Ubuntu.  If that works and Windows fails, let us know.  We'll investigate it as a Windows-specific issue.

**Reply by filippesic** [reply]
Just a quick update.. I tried:
`magick -monitor -limit memory 2GiB -limit map 4GiB -define registry:temporary-path=/media/sf_image_magick_test -density 231x231 DATA_FOLDER/*.tif image.pdf`
On Ubuntu in Virtual Box with just 5GB of shared memory since I only have 8GB on my home PC. It actually worked! I can view all pages in the .pdf file, all 331 of them work now. I checked in Windows Adobe Reader and everything is perfect. RAM consumption did not exceed 3.1GB in Ubuntu, and about 7.2GB in Windows while the Ubuntu and magick was running of course. `/media/sf_image_magick_test` or something like that is the shared Virtual Box folder for files that I needed to convert to pdf, so I used that folder as temp folder, because it's an SSD, ~70GB free. Also, how can I download -20 version? I looked everywhere..
>"In the mean-time we created a 5GB file with this command:"

That's okay, my output pdf file is 4.2GB, that's not the issue, the issue is that about 18 pages are blank, yet the file is 4.2GB. Now when I tried through Ubuntu the end pdf is also 4.2GB, but those 18 pages are working, it's not blank anymore.

**Reply by urban-warrior** [reply]
Apologies, that should be 7.0.10-19.

We ran your command successfully under Windows Bash.  We're now running under Windows PowerShell to see if we can replicate the issue.  We'll let you know our findings once the test completes.

**Reply by filippesic** [reply]
Oh, okay no problem. I've been using that version of course. I used Git Bash on Windows, but that shouldn't make a difference(tried CMD too). Since I tried this on my home PC, laptop on work and that VM with 24 cores and 47GB of RAM, then the problem must be with my .tif files. I get a lot of field tag errors, but I don't think that affects the converting. I forgot to mention that I also tried to convert 312 tif files to singe pdf than the rest(~18) of tif files to other pdf file and then add those 2 pdf files together. Same problem.

Here's a file info from one of the files: https://imgur.com/a/D4ozkJA

**Reply by fmw42** [reply]
You can suppress all the warning (not errors) using -quiet. Add that right after magick.
